Proton/Sources/Core/PRTextStorage.m

```diff
@@ -69,48 +63,44 @@ - (id)attribute:(NSAttributedStringKey)attrName atIndex:(NSUInteger)location eff
     return [self.storage attribute:attrName atIndex:location effectiveRange:range];
 }
 
-- (NSDictionary<NSString *,id> *)attributesAtIndex:(NSUInteger)location effectiveRange:(NSRangePointer)effectiveRange {
-//    if (location <= self.storage.length) {
-//        return [[NSDictionary alloc] init];
-//    }
-    
+- (NSDictionary<NSString *, id> *)attributesAtIndex:(NSUInteger)location effectiveRange:(NSRangePointer)effectiveRange {
     return [self.storage attributesAtIndex:location effectiveRange:effectiveRange];
 }
 
 - (void)replaceCharactersInRange:(NSRange)range withAttributedString:(NSAttributedString *)attrString {
-      // TODO: Add undo behaviour
-   NSMutableAttributedString *replacementString = [attrString mutableCopy];
+    // TODO: Add undo behaviour
+    NSMutableAttributedString *replacementString = [attrString mutableCopy];
     // Fix any missing attribute that is in the location being replaced, but not in the text that
     // is coming in.
     if (range.length > 0 && attrString.length > 0) {
-        id outgoingAttrs = [_storage attributesAtIndex:(range.location + range.length - 1) effectiveRange:nil];
-        id incomingAttrs = [attrString attributesAtIndex:0 effectiveRange:nil];
+        NSDictionary<NSAttributedStringKey, id> *outgoingAttrs = [_storage attributesAtIndex:(range.location + range.length - 1) effectiveRange:nil];
+        NSDictionary<NSAttributedStringKey, id> *incomingAttrs = [attrString attributesAtIndex:0 effectiveRange:nil];
 
-        NSMutableDictionary<NSAttributedStringKey,id> *diff = [[NSMutableDictionary alloc] init];
-        for (id outgoingKey in outgoingAttrs) {
+        NSMutableDictionary<NSAttributedStringKey, id> *diff = [[NSMutableDictionary alloc] init];
+        for (NSAttributedStringKey outgoingKey in outgoingAttrs) {
             // We do not want to fix the underline since it can be added by the input method for
             // characters accepting diacritical marks (eg. in Vietnamese or Spanish) and should be transient.
-            if (![incomingAttrs valueForKey:outgoingKey] && outgoingKey != NSUnderlineStyleAttributeName) {
```

Comment:
`valueForKey` is a KVC method. `NSDictionary` has `objectForKey` which could be replaced with (Object Subscripting)[https://nshipster.com/object-subscripting/]